docs/tutorials/mongoid-queries.txt

```diff
@@ -42,6 +42,78 @@ is returned with the new query added. This is so that with scoping
 or exposures, for example, the original queries are unmodified
 and remain reusable.
 
+elem_match
+``````````
+
+This matcher finds documents with array fields where one of the array values
+matches all of the conditions. For example:
+
+.. code-block:: ruby
+
+  class Band
+    include Mongoid::Document
+    field :name, type: String
+    field :tours, type: Array
+  end
+
+  aerosmith = Band.create!(name: 'Aerosmith', tours: [
+    {city: 'London', year: 1995},
+    {city: 'New York', year: 1999},
+  ])
+
+  Band.elem_match(tours: {city: 'London'}).to_a # => [aerosmith]
+
+``elem_match`` also works with embedded associations:
+
+.. code-block:: ruby
+
+  class Band
+    include Mongoid::Document
+    field :name, type: String
+    embeds_many :tours
+  end
+
+  class Tour
+    include Mongoid::Document
+    field :city, type: String
+    field :year, type: Integer
+    embedded_in :band
+  end
+
+  dm = Band.create!(name: 'Depeche Mode')
+  aerosmith = Band.create!(name: 'Aerosmith')
+  Tour.create!(band: aerosmith, city: 'London', year: 1995)
+  Tour.create!(band: aerosmith, city: 'New York', year: 1999)
+
+  Band.elem_match(tours: {city: 'London'}).to_a # => [aerosmith]
+
+``elem_match`` does not work with non-embedded associations because MongoDB
```

Comment:
Out of curiosity, do you think this is something we could implement with the `$lookup` aggregation stage? It seems to me like it might be doable.

Reply:
Converting a collection criteria to an aggregation pipeline is probably not a viable strategy, but I totally support documenting how to accomplish the equivalent operation with aggregation pipeline.